ida_launcher/src/ida_launcher.c

```diff
@@ -0,0 +1,88 @@
+#include <windows.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+// Function to search registry key for the highest version of IDA installation
+int findHighestIdaVersion(HKEY rootKey, LPCSTR subKey, LPCSTR idaIdentifier, LPSTR path, DWORD pathSize) {
+    HKEY hKey;
+    DWORD dwIndex = 0;
+    char keyName[260];
+    char displayName[260];
+    char highestVersionPath[260] = {0};
+    LONG lRes;
+    BOOL found = FALSE;
+
+    lRes = RegOpenKeyExA(rootKey, subKey, 0, KEY_READ, &hKey);
+    if (lRes != ERROR_SUCCESS) {
+        return 0; // Unable to open registry key
+    }
+
+    while (TRUE) {
+        DWORD dwKeyNameSize = sizeof(keyName);
+        lRes = RegEnumKeyExA(hKey, dwIndex, keyName, &dwKeyNameSize, NULL, NULL, NULL, NULL);
+        if (lRes != ERROR_SUCCESS) {
+            break; // No more keys or an error occurred
+        }
+
+        HKEY hSubKey;
+        lRes = RegOpenKeyExA(hKey, keyName, 0, KEY_READ, &hSubKey);
+        if (lRes == ERROR_SUCCESS) {
+            DWORD displayNameSize = sizeof(displayName);
+            lRes = RegQueryValueExA(hSubKey, "DisplayName", NULL, NULL, (LPBYTE)displayName, &displayNameSize);
+            if (lRes == ERROR_SUCCESS && strstr(displayName, idaIdentifier) != NULL) {
+                DWORD valueSize = pathSize;
+                char currentPath[260] = {0};
+                lRes = RegQueryValueExA(hSubKey, "InstallLocation", NULL, NULL, (LPBYTE)currentPath, &valueSize);
+                if (lRes == ERROR_SUCCESS) {
+                    // Compare versions and keep the highest one
+                    if (strcmp(currentPath, highestVersionPath) > 0) {
+                        strncpy(highestVersionPath, currentPath, sizeof(highestVersionPath));
+                        found = TRUE;
+                    }
+                }
+            }
+            RegCloseKey(hSubKey);
+        }
+        dwIndex++;
+    }
+
+    RegCloseKey(hKey);
+
+    if (found) {
+        strncpy(path, highestVersionPath, pathSize);
+    }
+
+    return found;
+}
+
+// WinMain entry point
+int WINAPI WinMain(HINSTANCE hInstance, HINSTANCE hPrevInstance, LPSTR lpCmdLine, int nCmdShow) {
+    char idaPath[260] = {0};
+    DWORD dwSize = sizeof(idaPath);
+
+    // Search for the highest version of IDA Pro first
+    if (!findHighestIdaVersion(HKEY_LOCAL_MACHINE, "SOFTWARE\\Wow6432Node\\Microsoft\\Windows\\CurrentVersion\\Uninstall", "IDA Pro", idaPath, dwSize) &&
```

Comment:
I would prefer to look for the IDA executables in `%programfiles%` instead of using the Uninstall registry key (similar to the current PowerShell script). I think the code would be shorter, simplier and easier to understand.

As `FindFirstFile` supports wildcharacters, I think `FindFirstFile("%programfiles%\\IDA Pro *\\ida64.exe", &FindFileData)` returns the highest IDA Pro version and `FindFirstFile("%programfiles%\\IDA Freeware *\\ida64.exe", &FindFileData)` returns the highest IDA free version.

Reply:
The issue I found with going the `%programfiles%` route was that a user can install IDA anywhere on the system, so this isn't guaranteed to work. Searching just within `%programfiles%` is an assumption we are making, whereas if IDA is installed, it should always be located in the registry (as far as I can tell).

If we're willing to stick with that assumption that IDA should -always- be installed inside of the default location (even IDA Pro for other users down the road), then I will make that change. I just want to confirm that we want to accept that tradeoff before I make the change. :)

Reply:
For this. I opted to stick with the registry, as a test with going the `FindFirstFile` route did not seem to be much cleaner. It also leads to issues with `strcmp` for versioning, where we need to deal with strings like `10.0` being less than `8.3` according to `strcmp` for example. 

While there are clearly ways we can make this work, I don't think the effort is worth it when the `registry` route is working currently, and now that it's been adjusted to look at versioning via the registry (Major and Minor) which uses numbers, the comparison is much more straightforward.